src/sqe.rs

```diff
@@ -173,6 +175,41 @@ impl<'a> SubmissionQueueEvent<'a> {
         self.sqe.flags = flags.bits() as _;
     }
 
+    /// Prepare an atomic vectored read using the `readv` system call.
+    ///
+    /// # Safety
+    /// You must ensure the file `fd` and destination buffer slice `bufs` outlive the event.
```

Comment:
The requirements could be stated more precisely. The user needs to treat these as if the kernel has exclusive access to them until the completion occurs - they can't read or write to the bufs array or the buffers it points to and they can't pass the fd to kernel any syscalls like read/write/close (or schedule another event on the fd that isn't temporally linked to this one (ie using `IOSQE_IO_LINK`)

So I would at least change this to something like "ensure the kernel has exclusive access to the file and destination buffers until the end of the event"

Reply:
Thank you, fixed in fbf39b3 :)
Was having a little trouble with the wording.